runtime/lua/vim/tshighlighter.lua

```diff
@@ -56,6 +55,19 @@ function TSHighlighter:set_query(query)
   end
   self.query = query
 
+  self.hl_cache = {}
+  for id, name in pairs(self.query.captures) do
+      local firstc = string.sub(name, 1, 1)
+      local hl
+      if firstc ~= string.lower(firstc) then
```

Comment:
I don't really understand the logic below.

Reply:
I added some comments.

Reply:
In the end I did not, what do you want me to explain here ?